silkworm/core/types/block.hpp

```diff
@@ -94,13 +96,17 @@ struct BlockBody {
     std::vector<BlockHeader> ommers;
     std::optional<std::vector<Withdrawal>> withdrawals{std::nullopt};
 
+    // EOS-EVM 
+    std::optional<uint64_t> consensus_parameter_index{std::nullopt};
+
     friend bool operator==(const BlockBody&, const BlockBody&);
 };
 
 struct Block : public BlockBody {
     BlockHeader header;
 
     bool irreversible{false};
+    std::optional<eosevm::ConsensusParameters> consensus_parameters_cache;
```

Comment:
Do we need to have this cache here?
I thought it will be local to the function that prepares the execution of the block

Reply:
This is because we are currently only passing block around: from block_conversion_plugin to blockchain_plugin, from blockchain_plugin to the executor

It will be safer to copy and pass this piece of information with block. otherwise we will have to let executor read the new parameters from somewhere, and that will start to encounter thread safety issues.

This approach might not be ideal, Feel free to suggest other approaches.